webrender/src/platform/windows/font.rs

```diff
@@ -85,98 +139,88 @@ impl FontContext {
-                                                      dwrote::DWRITE_TEXTURE_CLEARTYPE_3x1),
-        };
+        let dwrite_render_mode = self.get_dwrite_render_mode(render_mode);
+        let dwrite_measure_mode = self.get_dwrite_measure_mode(render_mode);
 
         // XX use the xform to handle subpixel positioning (what skia does), I believe that keeps
         //let xform = dwrote::DWRITE_MATRIX { m11: 1.0, m12: 0.0, m21: 0.0, m22: 1.0, dx: 0.0, dy: 0.0 };
-        let analysis = dwrote::GlyphRunAnalysis::create(&glyph_run, 1.0, None, r_mode, m_mode, 0.0, 0.0);
-        let bounds = analysis.get_alpha_texture_bounds(tex_type);
+        dwrote::GlyphRunAnalysis::create(&glyph_run, 1.0, None,
+                                         dwrite_render_mode,
+                                         dwrite_measure_mode,
+                                         0.0, 0.0)
+    }
+
+    fn get_glyph_dimensions_with_analysis(&self, analysis: dwrote::GlyphRunAnalysis,
+                                          texture_type: dwrote::DWRITE_TEXTURE_TYPE)
+                                         -> GlyphDimensions {
+        let bounds = analysis.get_alpha_texture_bounds(texture_type);
 
         let width = (bounds.right - bounds.left) as u32;
         let height = (bounds.bottom - bounds.top) as u32;
-        let dims = GlyphDimensions {
+        GlyphDimensions {
             left: bounds.left,
             top: -bounds.top,
             width: width,
             height: height,
-        };
-
-        // if empty, then nothing
-        if dims.width == 0 || dims.height == 0 {
-            return (None, None);
         }
+    }
 
-        // if we weren't asked to rasterize, we're done
-        if render_mode.is_none() {
-            return (Some(dims), None);
-        }
+    pub fn get_glyph_dimensions(&self,
+                                font_key: FontKey,
+                                size: Au,
+                                glyph: u32) -> Option<GlyphDimensions> {
+        // Probably have to default to something else here.
+        let render_mode = FontRenderMode::Subpixel;
+        let analysis = self.create_glyph_analysis(font_key, size, glyph, render_mode);
+
+        let texture_type = self.get_dwrite_texture_type(render_mode);
+        Some(self.get_glyph_dimensions_with_analysis(analysis, texture_type))
+    }
 
-        let pixels = analysis.create_alpha_texture(tex_type, bounds);
-        let rgba_pixels = match render_mode.unwrap() {
+    // DWRITE gives us values in RGB. WR expects it in RGBA. Note, CG returns in BGR
+    // TODO: probably should fix that
+    fn convert_to_rgba(&self, pixels: &Vec<u8>, render_mode: FontRenderMode) -> Vec<u8> {
+        match render_mode {
             FontRenderMode::Mono => {
-                let mut rgba_pixels = vec![0; pixels.len() * 4];
+                let mut rgba_pixels = vec![0 as u8; pixels.len() * 4];
```

Comment:
do you need the `as u8`? Doesn't hurt to be explicit, though if you want to be explicit I'd prefer `rgba_pixels: Vec<u8>`.  Don't feel strongly about this, though.